Interactive start pipeline throws error

# Expected Behavior

`tkn pipeline start` should work

# Actual Behavior

It throws error

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x131a278]

goroutine 1 [running]:
github.com/tektoncd/cli/pkg/cmd/pipeline.(*startOptions).getInputParams(0xc000129040, 0xc000202000, 0x1, 0x1)
	/home/pgarg/go/src/github.com/tektoncd/cli/pkg/cmd/pipeline/start.go:255 +0x138
github.com/tektoncd/cli/pkg/cmd/pipeline.(*startOptions).getInput(0xc000129040, 0x7ffcea1c4105, 0xd, 0x1916e01, 0xc000735b00)
	/home/pgarg/go/src/github.com/tektoncd/cli/pkg/cmd/pipeline/start.go:191 +0x132
github.com/tektoncd/cli/pkg/cmd/pipeline.(*startOptions).run(0xc000129040, 0x7ffcea1c4105, 0xd, 0x1916e20, 0xc000010020)
	/home/pgarg/go/src/github.com/tektoncd/cli/pkg/cmd/pipeline/start.go:157 +0x43
github.com/tektoncd/cli/pkg/cmd/pipeline.startCommand.func3(0xc0002db8c0, 0xc0002cefd0, 0x1, 0x1, 0x0, 0x0)
	/home/pgarg/go/src/github.com/tektoncd/cli/pkg/cmd/pipeline/start.go:137 +0x123
github.com/spf13/cobra.(*Command).execute(0xc0002db8c0, 0xc0002cefa0, 0x1, 0x1, 0xc0002db8c0, 0xc0002cefa0)
	/home/pgarg/go/pkg/mod/github.com/chmouel/cobra@v0.0.0-20191021105835-a78788917390/command.go:832 +0x460
github.com/spf13/cobra.(*Command).ExecuteC(0xc0000f6b00, 0xc000143590, 0xc0000f6b00, 0xc000050118)
	/home/pgarg/go/pkg/mod/github.com/chmouel/cobra@v0.0.0-20191021105835-a78788917390/command.go:920 +0x2fb
github.com/spf13/cobra.(*Command).Execute(...)
	/home/pgarg/go/pkg/mod/github.com/chmouel/cobra@v0.0.0-20191021105835-a78788917390/command.go:870
main.main()
	/home/pgarg/go/src/github.com/tektoncd/cli/cmd/tkn/main.go:31 +0x4b

```

# Steps to Reproduce the Problem

1. Create a pipeline which has param with no default
2. Start pipeline and provide param value using interactive
3. Get the error

# Additional Info

```
It fails when you have params with no default either string or array. 
Also if you have array param with a default value, it does not take that value.
```

/kind bug
/assign

--- a/pkg/cmd/pipeline/start.go
+++ b/pkg/cmd/pipeline/start.go
@@ -238,6 +238,7 @@ func (opt *startOptions) getInputResources(resources resourceOptionsFilter, pipe
 				return err
 			}
 			opt.Resources = append(opt.Resources, res.Name+"="+newres.Name)
+			continue
 		}
 		name := strings.TrimSpace(strings.Split(ans, " ")[0])
 		opt.Resources = append(opt.Resources, res.Name+"="+name)
@@ -247,14 +248,24 @@ func (opt *startOptions) getInputResources(resources resourceOptionsFilter, pipe
 
 func (opt *startOptions) getInputParams(pipeline *v1alpha1.Pipeline) error {
 	for _, param := range pipeline.Spec.Params {
-		var ans string
+		var ans, ques, defaultValue string
+		ques = fmt.Sprintf("Value for param `%s` of type `%s`?", param.Name, param.Type)
+		input := &survey.Input{}
+		if param.Default != nil {
+			if param.Type == "string" {
+				defaultValue = param.Default.StringVal
+			} else {
+				defaultValue = strings.Join(param.Default.ArrayVal, ",")
+			}
+			ques = ques + fmt.Sprintf(" (Default is `%s`)", defaultValue)
+			input.Default = defaultValue
+		}
+		input.Message = ques
+
 		var qs = []*survey.Question{
 			{
-				Name: "pipeline param",
-				Prompt: &survey.Input{
-					Message: fmt.Sprintf("Value of param `%s` ? (Default is %s)", param.Name, param.Default.StringVal),
-					Default: param.Default.StringVal,
-				},
+				Name:   "pipeline param",
+				Prompt: input,
 			},
 		}
 

--- a/pkg/cmd/pipeline/start_test.go
+++ b/pkg/cmd/pipeline/start_test.go
@@ -260,6 +260,7 @@ func Test_start_pipeline_interactive(t *testing.T) {
 					tb.PipelineDeclaredResource("git-repo", "git"),
 					tb.PipelineParamSpec("pipeline-param", v1alpha1.ParamTypeString, tb.ParamSpecDefault("somethingdifferent")),
 					tb.PipelineParamSpec("rev-param", v1alpha1.ParamTypeString, tb.ParamSpecDefault("revision")),
+					tb.PipelineParamSpec("array-param", v1alpha1.ParamTypeArray, tb.ParamSpecDefault("revision1", "revision2")),
 					tb.PipelineTask("unit-test-1", "unit-test-task",
 						tb.PipelineTaskInputResource("workspace", "git-repo"),
 						tb.PipelineTaskOutputResource("image-to-use", "best-image"),
@@ -302,7 +303,7 @@ func Test_start_pipeline_interactive(t *testing.T) {
 					return err
 				}
 
-				if _, err := c.ExpectString("Value of param `pipeline-param` ?"); err != nil {
+				if _, err := c.ExpectString("Value for param `pipeline-param` of type `string`? (Default is `somethingdifferent`)"); err != nil {
 					return err
 				}
 
@@ -310,11 +311,19 @@ func Test_start_pipeline_interactive(t *testing.T) {
 					return err
 				}
 
-				if _, err := c.ExpectString("Value of param `rev-param` ?"); err != nil {
+				if _, err := c.ExpectString("Value for param `rev-param` of type `string`? (Default is `revision`)"); err != nil {
 					return err
 				}
 
-				if _, err := c.SendLine("test2"); err != nil {
+				if _, err := c.SendLine("test1"); err != nil {
+					return err
+				}
+
+				if _, err := c.ExpectString("Value for param `array-param` of type `array`? (Default is `revision1,revision2`)"); err != nil {
+					return err
+				}
+
+				if _, err := c.SendLine("test2, test3"); err != nil {
 					return err
 				}
 

